File: backend/app/models/job.py
import uuid
from datetime import datetime

class Job:
    # In-memory storage (replace with database later)
    _jobs = {}

    def __init__(self, job_id, status, created_at=None, result_url=None, error=None):
        self.job_id = job_id
        self.status = status
        self.created_at = created_at or datetime.now()
        self.result_url = result_url
        self.error = error

    @classmethod
    def create(cls, job_id, status):
        job = cls(job_id, status)
        cls._jobs[job_id] = job
        return job

    @classmethod
    def get(cls, job_id):
        return cls._jobs.get(job_id)
    @classmethod
    def update(cls, job_id, status, result_url=None, error=None):
        job = cls._jobs.get(job_id)
        if job:
            job.status = status
            job.result_url = result_url
            job.error = error

File: project/lib/models/job.dart
// lib/models/job.dart

import 'dart:typed_data';
import 'package:project/constants/enums.dart';

/// This is your app-level “ProcessingJob” model. 
/// It no longer needs to include SQLite mapping (toMap/fromMap).
/// Persisted storage is handled via `ProcessingJobHive` and `HiveHelper`.
class ProcessingJob {
  final String jobId;
  final String userId;
  final JobStatus status;
  final DateTime createdAt;
  final Uint8List? resultImage;
  final String? resultUrl;
  final String? localImagePath;
  final String? error;
  final String? batchId;
  final String? message;

  ProcessingJob({
    required this.jobId,
    required this.userId,
    required this.status,
    required this.createdAt,
    this.resultImage,
    this.resultUrl,
    this.localImagePath,
    this.error,
    this.batchId,
    this.message,
  });

  /// Derived property: true if the job is done or failed.
  bool get isComplete =>
      status == JobStatus.completed || status == JobStatus.failed;

  /// Creates a copy of this job, overriding only the provided fields.
  ProcessingJob copyWith({
    String? jobId,
    String? userId,
    JobStatus? status,
    DateTime? createdAt,
    Uint8List? resultImage,
    String? resultUrl,
    String? localImagePath,
    String? error,
    String? batchId,
    String? message,
  }) {
    return ProcessingJob(
      jobId: jobId ?? this.jobId,
      userId: userId ?? this.userId,
      status: status ?? this.status,
      createdAt: createdAt ?? this.createdAt,
      resultImage: resultImage ?? this.resultImage,
      resultUrl: resultUrl ?? this.resultUrl,
      localImagePath: localImagePath ?? this.localImagePath,
      error: error ?? this.error,
      batchId: batchId ?? this.batchId,
      message: message ?? this.message,
    );
  }
}
